File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ans = INT_MAX;
        int sum=0;
        int start=0;
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            while(sum-nums[start]>=target){
                    sum-=nums[start];
                    start++;
            }
            if(sum>=target) ans=min(ans,i-start+1);
        }
        return ans==INT_MAX ? 0 : ans;

    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.ts
function minSubArrayLen(target: number, nums: number[]): number {
    const arr = []
    let sum = 0
    let ans = nums.length;
    for(let i=0;i<nums.length;i++){
        sum+=nums[i];
        arr.push(nums[i]);
        while(sum-arr[0]>=target){
            sum-=arr[0];
            arr.shift();
        }
        if(sum>=target) ans = Math.min(ans,arr.length);
    }
    return sum>=target ? ans : 0;
};